### ISSUE TITLE
Failed install on Redmi Note 7

### ISSUE THREAD

**Issue by chipotledonut** [Divested-Mobile/DivestOS-Build#233]
When I run 

`fastboot update divested-18.1-20230810-dos-lavender-fastboot.zip`

I get this error:

`Bootloader Version: 
Baseband Version: 
Serial Number: c1368f9

extracting android-info.txt (0 MB) to RAM...
Checking 'product'                                 FAILED

Device product is 'lavender'.
Update requires 'sdm660'.
`

I can't find any info about that.

**Comment by SkewedZeppelin** [comment]
Try: `fastboot update --force divested-18.1-20230810-dos-lavender-fastboot.zip`

If that doesn't work, then extract the zip and `fastboot flash` each partition, eg. `fastboot flash system system.img`

**Comment by chipotledonut** [comment]
> Try: `fastboot update --force divested-18.1-20230810-dos-lavender-fastboot.zip`
> 
> If that doesn't work, then extract the zip and `fastboot flash` each partition, eg. `fastboot flash system system.img`

I did that and now it is stuck in Fastboot. Can´t turn it off, can´t boot into recovery

`fastboot reboot recovery` it boots in fastboot mode

**Comment by SkewedZeppelin** [comment]
Ah, I forgot that the DivestOS recovery is broken on this device: https://divestos.org/pages/broken#lavender

Switch back to Orangefox and `adb sideload divested-18.1-20230810-dos-lavender.zip`

should work fine then

**Comment by chipotledonut** [comment]
> Switch back to Orangefox

I can´t. I tried to flash Orangefox again, but it wont boot to recovery. It boots to fastboot

**Comment by SkewedZeppelin** [comment]
So downloading the zip: https://orangefox.download/device/lavender
- extract it
- fastboot boot recovery.img
- it should boot orangefox
- adb sideload OrangeFox-lavender-stable@R11.1_5_A12.zip
- adb sideload divested-18.1-20230810-dos-lavender.zip
- reboot to recovery should be Orangefox and reboot to system should be DivestOS

**Comment by chipotledonut** [comment]
> * fastboot boot recovery.img

`Sending 'boot.img' (33804 KB)                      OKAY [  1.220s]
Booting                                            FAILED (remote: 'Failed to load/authenticate boot image: Load Error')
fastboot: error: Command failed
`
That's what I get.

Did i brick it?

**Comment by chipotledonut** [comment]
It doesn't have slots.

I tried restarting, another port...

I'm flashing stock and starting over.

Should I flash everything but recovery from here: divested-18.1-20230810-dos-lavender-fastboot.zip?

Or flash everything and then flash orangefox again?

**Comment by SkewedZeppelin** [comment]
after going back to stock the simplest would be installing orangefox and then sideloading the non-fastboot divestos zip

**Comment by chipotledonut** [comment]
what about all of this?

Reboot to the bootloader: With the device powered off, hold Volume Down + Power.
$ fastboot erase avb_custom_key
$ fastboot flash avb_custom_key avb_pkmd-lavender.bin
Reboot to the bootloader
$ fastboot update divested-18.1-20230810-dos-lavender-fastboot.zip

**Comment by SkewedZeppelin** [comment]
the avb key has no use on this device since it can't be relocked, so you can skip those steps
and using while using fastboot.zip is the proper step for first install, the recovery issue and requirement of orangefox makes it unnecessary

**Comment by chipotledonut** [comment]
The thing was that after running

`fastboot update --force divested-18.1-20230810-dos-lavender-fastboot.zip`

the phone get locked by Xiaomi accounts.

`Booting                                            FAILED (remote: 'Failed to load/authenticate boot image: Load Error') `

When I flashed MIUI ROM I got asked to activate my Xiaomi Account to unlock my device. I don´t even rememebr my account since I bough the phone in 2019 and flashed a custom ROM two weeks later.

I can´t get access to my account, so I can´t get my phone back. It is virtually bricked.

**Comment by SkewedZeppelin** [comment]
Are you able to use another account maybe?

Very strange you can't flash DivestOS, as it has numerous update checks from on last months and this months update, so it does work.

I wonder if maybe you're on an older or maybe newer firmware version that is incompatible? or have an incompatible model somehow?

**Comment by ghostiy** [comment]
@chipotledonut what did you do? are you able to flash it?

**Comment by SkewedZeppelin** [comment]
@ghostiy 
are you encountering this issue too?

**Comment by ghostiy** [comment]
@SkewedZeppelin 

yes i'm trying to flash DivestOs on the same device, I followed all prerequisite steps & unlocked developer options. 
which is listed in this website: https://divestos.org/builds/LineageOS/lavender/install.html

Skipped step 5,6, as you said earlier because it won't make any difference since the device can't be relooked. 
Proceeded with step 7: $ fastboot update divested-18.1-20230810-dos-lavender-fastboot.zip
and it didn't work. 

gives me this error: 
`--------------------------------------------
Bootloader Version...: 
Baseband Version.....: 
Serial Number........: e54d4c4
--------------------------------------------
extracting android-info.txt (0 MB) to RAM...
Checking 'product'                                 FAILED

Device product is 'lavender'.
Update requires 'sdm660'.

fastboot: error: requirements not met!
`

**Comment by ghostiy** [comment]
@SkewedZeppelin 

i even tried --force option.. 

and same error occured

**Comment by ghostiy** [comment]
@SkewedZeppelin

Btw, i have lineageOs installed as previous ROM, can i flash the zip rom recovery? 
i've 2 files downloaded for my device from the website: 
divested-18.1-20230810-dos-lavender.zip
divested-18.1-20230810-dos-lavender-fastboot.zip

**Comment by ghostiy** [comment]
Hi Again @SkewedZeppelin

I've rebooted into lineage recovery and sideloaded the DivestOs.zip file
there were some errors about signature i clicked ignore to cotinue. 

It then boot looped till i wiped data from recovery
now when it boots it only shows DivestOs splash screen for about 15 minutes.. And nothing happened, should iw ait longer?

**Comment by SkewedZeppelin** [comment]
@ghostiy 
no, it should only be 5 minutes max
sounds broken

**Comment by ghostiy** [comment]
@SkewedZeppelin 

yeah it's on the boot animation divestos splash screen.. 

maybe the problem was in the lineageos recovery? 
i will try to flash ofox then flash the .zip again with sideload.

**Comment by ghostiy** [comment]
@SkewedZeppelin 

it worked. i'm now into system and it's running good. thanks.
am i missing anything from not applying the `fastboot update divested-18.1-20230810-dos-lavender-fastboot.zip` step?

**Comment by SkewedZeppelin** [comment]
@ghostiy 
nah, if it is booted you are good.

happy to hear it indeed works

**Comment by ghostiy** [comment]
@SkewedZeppelin 
Thank you & all divestos team members for your work <3

**Comment by ghostiy** [comment]
@SkewedZeppelin 

i'm sorry for so many questions. 
but any plans for future Android upgrade for this device?

**Comment by chipotledonut** [comment]
So,  @ghostiy got the same error than me. Fortunetly for him, he didnt follow any of the official steps and just flashed the ROM like any other ROM.

I followed official steps and advice here and got a semi brick phone.

**Comment by chipotledonut** [comment]
Theres something wrong in this ROM for this device and it would be good to either mention it in its page or drop support. 

> Device product is 'lavender'.
> Update requires 'sdm660'.

Everyone gets this error and from there, following official workarounds like --force and so on is the beginning of a lot of trouble

**Comment by ghostiy** [comment]
@chipotledonut 

Honestly, the instructions on the website are confusing, not straight forward, not device specified... 
it's kinda general for all the devices. and it's up to your luck if you figure how to make it work for you or not

luckily, the `--force` option didn't work for me at the first time.. And i don't know why.

just try to flash another rom and hope it will fix the issue, maybe another recovery too. wipe MIUI .. 
If you've unlocked the phone before using their tool. i think you're ok with anything else

**Comment by chipotledonut** [comment]
> just try to flash another rom

I couldnt boot into any recovery. I couldn install any recovery. Only thing that worked was to flash MIUI again BUT of course I had to unlocke my phone with my Mi Account. I bought the phone in 2019. Only god know what my Mi account is. My fault, I guess. I could talk to Xiaomi to try to get this solved but honestly, they wouldnt give a fuck.

Anyways, I bought a Pixel,  GrapheneOS,  not looking back again to unlock Xiaomi phones or similar.

**Comment by ghostiy** [comment]
I'm sorry for your bad experience.. 
maybe you can try with forgot password option on their site or with your phone number you used to register your phone with? 

Here, i think this link might help you with your problem, but you can try.  
[https://techrushi.com/solve-this-device-is-locked-in-xiaomi-phones/](url)
> Anyways, I bought a Pixel, GrapheneOS, not looking back again to unlock Xiaomi phones or similar.

yeah, that's the best choice of phone brand. congrats on your new phone. 
but you can still try on lavender.. good for use for anything else.

i think i will be forever loyal to lavender tho..  until it dies at least lol

**Comment by brodclaw** [comment]
I had same issues. I managed to unbrick it by flashing stock MIUI with Mi Flash Tool back if I remember correctly. Only way I could make this ROM work is if I flashed it as any other ROM, using OrangeFox recovery.

**Comment by SkewedZeppelin** [comment]
I did adjust the guide for this: https://divestos.org/builds/LineageOS/lavender/install.html
